include/rocksdb/utilities/backupable_db.h

```diff
@@ -110,6 +110,12 @@ struct BackupableDBOptions {
 
   // When Open() is called, it will open at most this many of the latest
   // non-corrupted backups.
+  //
+  // Note setting this to a non-default value prevents old files from being
+  // deleted in certain cases, as we can't do proper ref-counting. If using this
+  // option, make sure to occasionally disable it (by resetting to INT_MAX) and
+  // run GarbageCollect to clean accumulated stale files.
```

Comment:
Could you also elaborate on the cases in which old files will not be deleted? 
Also, its possible that once an option is set, users rarely re-visit it as long as it is working well. So I think it is too much ask our users to remember to reset and GarbageCollect once in a while :)

Reply:
The use case of this option is if you have certain BackupEngines that only intend to write new backups. Then you can set this to zero and it won't fail transiently/lag for a long time trying to open irrelevant old backups. The problem, though, is such BackupEngines can't determine which files are obsolete so can't properly clean up old backups. If people want to use this feature and get proper cleanup, they need to create some BackupEngines without this option set.

I am not sure if there's a better way. Suggestions welcome :).

Reply:
Let me land this since right now it'd silently drop files, which is worse than being difficult to use. I hope to deprecate this feature in RocksDB 6.0 as it has fundamental problems with garbage collection that I didn't foresee.